File: GuessTheNumber.cpp
#include <iostream>
#include <conio.h>

int main(void)
{
	char username[10000];
	srand((unsigned)time(NULL));
	int randnum = rand() % 20 + 1; // (1,20)
	int usernum = -1;
	cout << "Hello! What is your name?" << endl;
	cin >> username;
	cout << "Well, " << username << ", I am thinking of a number between 1 and 20" << endl;
	int trynum = 1;

	while(randnum != usernum && trynum <= 6)
	{
		cout << "Take a guess" << endl;
		cin >> usernum;
		if(usernum < randnum){
			cout << "Your guess is too low" << endl;
		}
		else if(usernum > randnum){
			cout << "Your guess is too high" << endl;
		}
		else if(usernum == randnum){
			break;
		}
		trynum++;
	}
	if(usernum == randnum) {
		cout << "Good job, " << username << "! you guessed my number in " << trynum << " guesses!" << endl;
	}
	else if(usernum != random) {
		cout << "Nope. The number I was thinking of was " << randnum << endl;
	}
}

File: GuessTheNumber.py
import random

print 'Hello! What is your name?'
username = raw_input()
print 'Well, %s, I am thinking of a number between 1 and 20' % username
# print 'Well, ' + username + ', I am thinking of a number between 1 and 20'
answer = random.randint(1,20)
# print answer
input_answer = -1
count = 0
while answer != input_answer :
	print 'Take a guess.'
	input_answer = int(input())
	count += 1
	if input_answer > answer :
		print 'Your guess is too high.'
	elif input_answer < answer :
		print 'Your guess is too low.'
	elif input_answer == answer :
		print 'Good job, %s! You guessed my number in %d guesses!' % (username, count)
		# print 'Good job, ' + username + '! You guessed my number in ' + str(count) + ' guesses!'
